components/download_images/src/main.py

```diff
@@ -123,35 +141,43 @@ def __init__(self,
             max_aspect_ratio=max_aspect_ratio,
         )
 
-    def transform(
-            self,
-            dataframe: dd.DataFrame,
-    ) -> dd.DataFrame:
-        logger.info("Instantiating resizer...")
-
-        # Remove duplicates from laion retrieval
-        dataframe = dataframe.drop_duplicates()
-
-        dataframe = dataframe.apply(
-            lambda example: download_image_with_retry(
-                url=example.images_url,
-                timeout=self.timeout,
-                retries=self.retries,
-                resizer=self.resizer,
-            ),
-            axis=1,
-            result_type="expand",
-            meta={0: bytes, 1: int, 2: int},
+    def transform(self, dataframe: dd.DataFrame) -> dd.DataFrame:
+
+        logger.info(f"Length of the dataframe: {len(dataframe)}")
+        logger.info("Downloading images...")
+
+        # drop width and height columns, as those are going to be
+        # added later on
+        dataframe = dataframe.drop(columns=['images_width', 'images_height'])
+
+        # create meta
+        # needs to be a dictionary with keys = column names, values = dtypes of columns
+        # for each column in the output
+        meta = dict(zip(dataframe.columns, dataframe.dtypes))
+        meta["data"] = bytes
+        meta["width"] = int
+        meta["height"] = int
+
+        dataframe = dataframe.map_partitions(
```

Comment:
Is there still a reason to use a `DaskTransformComponent` for this component? If we use the `PandasTransformComponent`, we get the `map_partitions` for free.